app/models/entity.rb

```diff
@@ -1,4 +1,9 @@
 class Entity < ApplicationRecord
-  belongs_to :user
-  has_many :groups
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+
+  has_many :entities
+  has_many :groups, through: :entities
```

Comment:
- [ ] Kindly note that is no foreig_keys in your tables that certify these associations you made. kindly run a migration that will certify the association you made in the above code. this is leading your app to some performance issues(infinite loops) 

![image](https://user-images.githubusercontent.com/65068771/211952448-252462c4-49a4-4a8b-ab3b-6453149f5c32.png)

Reply:
done